R/prompt.R

```diff
@@ -0,0 +1,53 @@
+#' Helpers for interpolating data into prompts
+#'
+#' These functions are lightweight wrappers around
+#' [glue](https://glue.tidyverse.org/) that make it easier to interpolate
+#' dynamic data into a static prompt. Compared to glue, these functions
+#' expect you to wrap dynamic values in `{{ }}`, making it easier to include
+#' R code and JSON in your prompt.
+#'
+#' @param prompt A prompt string
+#' @param ... Define additional temporary variables for substitution.
+#' @param .envir Environment to evaluate `...` expressions in. Used when
+#'   wrapping in another function. See `vignette("wrappers", package = "glue")`
+#'   for more details.
+#' @export
+#' @examples
+#' joke <- "You're a cool dude who loves to make jokes. Tell me a joke about {{topic}}."
+#'
+#' # You can supply valuese directly:
+#' prompt(joke, topic = "bananas")
+#'
+#' # Or allow prompt to find them in the current environment:
+#' topic <- "applies"
+#' prompt(joke)
+prompt <- function(prompt, ..., .envir = parent.frame()) {
+  check_string(prompt)
+
+  if (!dots_named(...)) {
+    cli::cli_abort("All elements of `...` must be named")
+  }
+
+  out <- glue::glue(prompt, ..., .open = "{{", .close = "}}", .envir = .envir)
```

Comment:
How about `glue_safe` here? At least by default? Otherwise we're making `.md` a pretty surprising attack vector.

Reply:
Can you give me a bit more detail on the attack vector that you're thinking of? I was thinking that `prompt` would usually be supplied by the code author, so they're the only ones who get to decide what's evaluated.